src/Compilers/VisualBasic/Portable/Utilities/FeatureUtils.vb

```diff
@@ -0,0 +1,84 @@
+            Dim required = feature.GetLanguageVersion()
+            Dim current = options.LanguageVersion
+            Return required <= current
+        End Function
+
+        <Runtime.CompilerServices.Extension>
+        Private Function CheckFeatures(feature As Feature, options As VisualBasicParseOptions) As Boolean
+            Dim featureFlag = feature.GetFeatureFlag()
+            Return ((featureFlag IsNot Nothing) AndAlso options.Features.ContainsKey(featureFlag))
+        End Function
+
+        ''' <summary>Report the unavailibility of a language feature.</summary>
+        ''' <typeparam name="TNode"></typeparam>
+        ''' <param name="feature">Language feature to report as unavailable.</param>
+        ''' <param name="options">The parse options being used.</param>
+        ''' <param name="node">The node to attach the diagnostic (Feature Unavailable).</param>
+        ''' <returns>Return the node with the diagnostic attached to it.</returns>
+        <Runtime.CompilerServices.Extension>
+        Public Function ReportFeatureUnavailable(Of TNode As VisualBasicSyntaxNode)(feature As Feature, options As VisualBasicParseOptions, node As TNode) As TNode
+            Dim featureName = ErrorFactory.ErrorInfo(feature.GetResourceId())
+            Dim requiredVersion As New VisualBasicRequiredLanguageVersion(feature.GetLanguageVersion())
+            Return Parser.ReportSyntaxError(node, ERRID.ERR_LanguageVersion, options.LanguageVersion.GetErrorName(), featureName, requiredVersion)
+        End Function
+
+        ''' <summary>
+        ''' Check to see if the given <paramref name="feature"/> is available with the <see cref="LanguageVersion"/>
+        ''' of the parser.
+        ''' </summary>
+        ''' <param name="feature">Language feature to check is available.</param>
+        ''' <param name="options">The parse options being used.</param>
+        ''' <returns></returns>
+        <Runtime.CompilerServices.Extension>
+        Public Function IsAvailable(feature As Feature, options As VisualBasicParseOptions) As Boolean
+            Return feature.CheckFeatures(options) OrElse feature.IsInLanguageVersion(options)
+        End Function
+
+        ''' <summary>
+        '''  Check to see if the given <paramref name="feature"/> is available with the <see cref="LanguageVersion"/>
+        ''' of the parser.  If it is not available a diagnostic will be added to the returned value.
+        ''' </summary>
+        ''' <param name="node">The node to attach the potential diagnostic (Feature Unavailable).</param>
+        ''' <param name="feature">Language feature to check is available.</param>
+        ''' <param name="options">The parse options being used.</param>
+        ''' <returns></returns>
+        <Runtime.CompilerServices.Extension>
+        Public Function CheckFeatureAvailable(Of TNode As VisualBasicSyntaxNode)(node As TNode, feature As Feature, options As VisualBasicParseOptions) As TNode
```

Comment:
Please use the original name `CheckFeatureAvailability`, for consistency with previous code and with the C# implementation.